mmv1/api/resource.go

```diff
@@ -492,6 +492,24 @@ func (r Resource) UserParameters() []*Type {
 	})
 }
 
+func (r Resource) ServiceVersion() string {
+	if r.CaiBaseUrl != "" {
+		return extractVersionFromBaseUrl(r.CaiBaseUrl)
+	}
+	return extractVersionFromBaseUrl(r.BaseUrl)
+}
+
+func extractVersionFromBaseUrl(baseUrl string) string {
```

Comment:
Could you add tests for this? (Possibly it could be combined with the other version extraction function?)